### PR TITLE
Reading gold token

### PR DESCRIPTION
**Pull request by mrsmkl**
Updates the gold token each time the native coin is updated. 

Also fixes a problem with "Celo info" tab always appearing when in "Blocks validated" tab.

See issue https://github.com/celo-org/celo-monorepo/issues/1988

Got a deadlock error at one phase, going to need more testing to find out if that is a problem.

### CODE REVIEW

**Comment by mrsmkl** [comment]
Seems to work on baklava-staging

**Comment by nambrot** [comment]
@mrsmkl Would you mind expanding on what do you think the problem, and highlight the solution here? From the changes, it's not readily apparent to me what the situation here is

**Comment by mrsmkl** [comment]
Yes, the native transfers aren't taken into account when handling the Gold token values, so it will only update when an erc-20 transaction is made. The change here is that each time an address is updated (a transaction that might change balance),it's added to the list of addresses where the Gold token balance should be updated too.

File: apps/explorer/lib/explorer/chain.ex [hunk 1]
```diff
@@ -886,6 +886,7 @@ defmodule Explorer.Chain do
             :contracts_creation_internal_transaction => :optional,
             :names => :optional,
             :smart_contract => :optional,
+            :celo_account => :optional,
```
Comment (mrsmkl) [inline apps/explorer/lib/explorer/chain.ex]: Fixes problem with Celo info tab,

File: apps/indexer/config/test.exs [hunk 2]
```diff
@@ -4,6 +4,8 @@ config :indexer, Indexer.Tracer, disabled?: false
 
 config :indexer, Indexer.Fetcher.CeloValidatorHistory.Supervisor, disabled?: true
 
+config :indexer, Indexer.Block.Fetcher, enable_special_token: false
```
Comment (mrsmkl) [inline apps/indexer/config/test.exs]: Disable extra updating of Gold token to make tests run.

File: apps/indexer/lib/indexer/block/fetcher.ex [hunk 3]
```diff
@@ -177,7 +209,13 @@ defmodule Indexer.Block.Fetcher do
              logs: logs,
              mint_transfers: mint_transfers,
              token_transfers: token_transfers,
-             transactions: transactions_with_receipts
+             transactions: transactions_with_receipts,
+             special_token:
+               if special_token_enabled do
+                 [%{hash: gold_token, block_number: last_block}]
```
Comment (mrsmkl) [inline apps/indexer/lib/indexer/block/fetcher.ex]: The address of the Gold token has to be added to the `addresses` table

File: apps/indexer/lib/indexer/block/fetcher.ex [hunk 4]
```diff
@@ -191,7 +229,14 @@ defmodule Indexer.Block.Fetcher do
            beneficiary_params_set
            |> add_gas_payments(transactions_with_receipts)
            |> BlockReward.reduce_uncle_rewards(),
-         address_token_balances = AddressTokenBalances.params_set(%{token_transfers_params: token_transfers}),
+         address_token_balances_from_transfers =
+           AddressTokenBalances.params_set(%{token_transfers_params: token_transfers}),
+         address_token_balances =
```
Comment (mrsmkl) [inline apps/indexer/lib/indexer/block/fetcher.ex]: Here all the modified addresses are added to the list for which the new token balances will be checked.

**Pull request merged**